libafl/src/bolts/shmem.rs

```diff
@@ -35,7 +35,7 @@ pub type StdShMemService = ShMemService<AshmemShMemProvider>;
 #[cfg(all(feature = "std", any(target_os = "ios", target_os = "macos")))]
 pub type StdShMemProvider = RcShMemProvider<ServedShMemProvider<MmapShMemProvider>>;
 #[cfg(all(feature = "std", any(target_os = "ios", target_os = "macos")))]
-pub type StdShMem = RcShMem<ServedShMem<MmapShMem>>;
+pub type StdShMem = RcShMem<ServedShMemProvider<MmapShMemProvider>>;
```

Comment:
This looks wrong. I don't get how this could even work, that way StdShMem would be a `ShMemProvider` instead of `ShMem`?
Maybe we can get rid of the `StdShMem` instead, and people should just use `StdShMemProvider::Mem` ?

Reply:
But here https://github.com/AFLplusplus/LibAFL/blob/15c6e6b73b551947cf9b97e22ba393a5caed2b6e/libafl/src/bolts/shmem.rs#L265

```
pub struct RcShMem<T: ShMemProvider> {
    internal: ManuallyDrop<T::Mem>,
    provider: Rc<RefCell<T>>,
}
``
RcShMem<T> wants T to be a `ShMemProvider` not a `ShMem`...
We can't just pass a `ShMem` into `RcShMem`, right?

Reply:
Ah you are correct